doc/usage.md

```diff
@@ -0,0 +1,22 @@
+# Picoquic CLI Usage
+On Linux after building, the following executables are available in the project root directory. All support "-h" argument to get the detailed usage:
+
+* picohttp_ct: runs various HTTP tests
+* picoquic_ct: runs various QUIC tests
+	* with no arguments, runs all tests.
+* picoquicdemo: QUIC/HTTP demo client and server
+* picoquic_sample: QUIC/HTTP demo client and server. See sample/README.md for detailed usage
+
+## HTTP CLI
+* client: ./picoquicdemo servername serverportnumber HTTPpath
+  * downloaded files will be in current directory. use -o for another folder 
+  * exemple: ./picoquicdemo -o ../received 192.0.2.1 4433 index.html
+* server: ./picoquicdemo -p portnumber
+  * -p argument tells the program that it is acting as a server
+  * HTML root folder is current directory by default. use -w for another folder
+  * exemple: ./picoquicdemo -w ../htmlroot -p 4433
+
```

Comment:
Technically, this is a "demo program", not just an "http cli". It supports the following application protocols:

* HTTP/3 -- the version of HTTP evolved to take advantage if QUIC, ALPN="h3"
* HTTP/0.9 -- the original version of HTTP, characterized by ALPN="hq-interop". this version is used in a number of introperability tests, e.g., [quic interop runner](https://interop.seemann.io/)
* QUIC Performance tests -- a simple protocol used to test performance of QUIC applications, ALPN="perf"
* Siduck -- a simple ping-pong protocol used to test support for datagrams, ALNP="siduck-00"

Reply:
I wanted something very short, straight to the point, for people to try it out. So I made some changes in the new commit, but did not go deep as you wrote. I would suggest that you put that very useful information in some other document describing the internals. But this is your repo, so you decide!